spyder/widgets/variableexplorer/arrayeditor.py

```diff
@@ -760,8 +760,11 @@ def change_active_widget(self, index):
         stack_index = self.dim_indexes[self.last_dim].get(data_index)
         if stack_index == None:
             stack_index = self.stack.count()
-            self.stack.addWidget(ArrayEditorWidget(self,
-                                                   self.data[slice_index]))
+            try:

```

Comment:
@ccordoba12 @Prikers isn't it better to check if it is empty instead of catching an index error?

```
NumPy's main object is the homogeneous multidimensional array. In Numpy dimensions are called axes. The number of axes is rank. Numpy's array class is called ndarray. It is also known by the alias array. The more important attributes of an ndarray object are:

ndarray.ndim
the number of axes (dimensions) of the array. In the Python world, the number of dimensions is referred to as rank.

ndarray.shape
the dimensions of the array. This is a tuple of integers indicating the size of the array in each dimension. For a matrix with n rows and m columns, shape will be (n,m). The length of the shape tuple is therefore the rank, or number of dimensions, ndim.

ndarray.size
the total number of elements of the array. This is equal to the product of the elements of shape.
```